server/api/router/auth.js

```diff
@@ -1,20 +1,71 @@
-const router = require('express').Router();
-const { User } = require('../../db/index');
-
-router.post('/login', (req, res, next) => {
-  const { email, password } = req.body;
-  User.findAll({ where: { email: email, password: password } })
-    .then(user => {
-      if (user) {
-        req.session.userId = user.id;
-        res.json(user);
-      } else {
-        const error = new Error('Wrong password or email!');
-        error.status = 401;
-        next(err);
-      }
-    })
-    .catch(next);
-});
-
-module.exports = router;
+const router = require('express').Router();
+const { User } = require('../../db/index');
+
+router.post('/login', async (req, res) => {
+  try {
+      const { email, password } = req.body;
+      if (email && password) {
+          const user = await User.findOne({ where: { email, password } });
+              if (!user) {
+                  res.sendStatus(401);
+                  console.log('Log in failed; invalid credentials');
+              } else if (user.loggedIn) {
+                res.sendStatus(200);
+                console.log('Already logged in as:', user.id);
+                } else {
+                const loggedInUser = await user.update({loggedIn: true});
```

Comment:
This is a bit strange... I wouldn't track if the user is logged in on the user model. **A user is logged in if their current session has a userId.**

So this should really be something like:

```javascript
const userSession = await session.findByPk(req.session.id);
await userSession.update({ userId: user.id });
req.session.userId = user.id;
```

This gets everything we could want accomplished done. But there is one issue... **If a user/client deleted their cookie, and is resigning into the site, they already have a session with their userId in our database!** This means that a user can have multiple authenticated sessions... This might not be a problem depending on how the rest of your codebase and relations are setup, but you can always:

1. Delete the old session here.
2. Tag sessions with the device type.

I don't know that you need to implement this as part of this PR, but you **do need to update away from the `user.update`.**